solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementFactory.java

```diff
@@ -242,6 +260,37 @@ public PlacementPlan computePlacement(Cluster cluster, PlacementRequest request,
         }
       }
 
+      if (secondaryCollection != null) {
+        // 2nd phase to allocate required secondary collection replicas
+        Set<Node> secondaryNodes = new HashSet<>();
+        Shard shard1 = secondaryCollection.iterator().next();
+        shard1.replicas().forEach(r -> {
+          secondaryNodes.add(r.getNode());
+        });
+        Set<ReplicaPlacement> secondaryPlacements = new HashSet<>();
+        Set<Node> alreadyAdded = new HashSet<>();
```

Comment:
Can add the nodes directly into `secondaryNodes` and save introducing `alreadyAdded`, given the actual placement decisions are captures elsewhere in `secondaryPlacements`.